### PR TITLE
SW-531_allow_optional_values

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

**Comment by TheKeymaster** [comment]
Tests run properly locally. Once this is merged, the test suite will automatically re-run and fail, since SW-530 is not yet merged into `develop` and even if it was, the changes of the `develop` branch must then be applied to `SW-481_support_supplying_product_id_for_debugging`.

File: tests/Export/XmlExportTest.php [hunk 1]
```diff
@@ -77,6 +77,15 @@ public function testProductsInCrossSellCategoriesAreNotWrappedAndErrorIsLogged()
...
     }
 
+    public function testKeywordsAreNotRequired(): void
+    {
+        $product = $this->createVisibleTestProduct(['tags' => []]);
+        $items = $this->getExport()->buildItems([$product], self::VALID_SHOPKEY, []);
+
+        $this->assertCount(1, $items);
+        $this->assertSame($product->getId(), $items[0]->getId());
+    }
```
Comment (dblaichinger) [inline tests/Export/XmlExportTest.php:87]: Please also assert that the keywords are empty.

File: tests/Export/XmlProductTest.php [hunk 2]
```diff
@@ -159,15 +160,34 @@ public function testAttributeException(): void
...
+    {
+        $product = $this->createVisibleTestProduct(['tags' => []]);
+
+        $xmlProduct = $this->getDefaultXmlProduct($product);
+        $xmlProduct->buildXmlItem();
+
+        $xmlItem = $xmlProduct->getXmlItem();
+
+        $this->assertNotNull($xmlItem);
+        $this->assertSame($product->getId(), $xmlItem->getId());
```
Comment (dblaichinger) [inline tests/Export/XmlProductTest.php:177]: Please also assert that the keywords are empty.

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: src/Export/XmlProduct.php [hunk 3]
```diff
@@ -140,43 +141,33 @@ private function setPrices(array $prices): void
...
-        }
+        $this->xmlItem->addDescription($description);
     }
 
     private function setDateAdded(?DateAdded $dateAdded): void
     {
-        if ($this->findologicProduct->hasDateAdded()) {
-            $this->xmlItem->setDateAdded($dateAdded);
-        }
+        $this->xmlItem->setDateAdded($dateAdded);
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Export/XmlProduct.php:149]: This comes from libflexport.

File: tests/Export/XmlExportTest.php (hunk 1 above)
Comment (TheKeymaster) [reply tests/Export/XmlExportTest.php:87]: I tried doing this, but libflexport does not provide any way to check this.

image

I would need to generate the item with an exporter and then manually parse that XML string.

File: tests/Export/XmlProductTest.php (hunk 2 above)
Comment (TheKeymaster) [reply tests/Export/XmlProductTest.php:177]: See [this comment](https://github.com/findologic/plugin-shopware-6/pull/124/files#r520412819).

File: tests/Export/XmlExportTest.php (hunk 1 above)
Comment (dblaichinger) [reply tests/Export/XmlExportTest.php:87]: No `getKeywords`? Libflexport would need a improvement :) Let's keep it at the id then.

**Review by dblaichinger — APPROVED** [review]

**Comment by TheKeymaster** [comment]
Closed in favor of https://github.com/findologic/plugin-shopware-6/pull/130

**Pull request closed without merging**